asset-transfer-basic/application-typescript/src/app.ts

```diff
@@ -0,0 +1,169 @@
+//         npm install
+// - to run this test application
+//         ===> from directory /fabric-samples/asset-transfer-basic/application-typescript
+//         npm start
+
+// NOTE: If you see  kind an error like these:
+/*
+    2020-08-07T20:23:17.590Z - error: [DiscoveryService]: send[mychannel] - Channel:mychannel received discovery error:access denied
+    ******** FAILED to run the application: Error: DiscoveryService: mychannel error: access denied
+
+   OR
+
+   Failed to register user : Error: fabric-ca request register failed with errors [[ { code: 20, message: 'Authentication failure' } ]]
+   ******** FAILED to run the application: Error: Identity not found in wallet: appUser
+*/
+// Delete the /fabric-samples/asset-transfer-basic/application-typescript/wallet directory
+// and retry this application.
+//
+// The certificate authority must have been restarted and the saved certificates for the
+// admin and application user are not valid. Deleting the wallet store will force these to be reset
+// with the new certificate authority.
+//
+
+/**
+ *  A test application to show basic queries operations with any of the asset-transfer-basic chaincodes
+ *   -- How to submit a transaction
+ *   -- How to query and check the results
+ *
+ * To see the SDK workings, try setting the logging to show on the console before running
+ *        export HFC_LOGGING='{"debug":"console"}'
+ */
+async function main() {
+    try {
+        // build an in memory object with the network configuration (also known as a connection profile)
+        const ccp = buildCCPOrg1();
+
+        // build an instance of the fabric ca services client based on
+        // the information in the network configuration
+        const caClient = buildCAClient(ccp, 'ca.org1.example.com');
+
+        // setup the wallet to hold the credentials of the application user
+        const wallet = await buildWallet(walletPath);
+
+        // in a real application this would be done on an administrative flow, and only once
+        await enrollAdmin(caClient, wallet, mspOrg1);
+
+        // in a real application this would be done only when a new user was required to be added
+        // and would be part of an administrative flow
+        await registerAndEnrollUser(caClient, wallet, mspOrg1, org1UserId, 'org1.department1');
+
+        // Create a new gateway instance for interacting with the fabric network.
+        // In a real application this would be done as the backend server session is setup for
+        // a user that has been verified.
+        const gateway = new Gateway();
+
+        try {
+            // setup the gateway instance
+            // The user will now be able to create connections to the fabric network and be able to
+            // submit transactions and query. All transactions submitted by this gateway will be
+            // signed by this user using the credentials stored in the wallet.
+            await gateway.connect(ccp, {
+                wallet,
```

Comment:
Could we create the "GatewayOptions" object before calling the "connect", I think it would be good to show how to set the different options in TypeScript

export interface GatewayOptions {
	identity: string | Identity;
	wallet?: Wallet;
	identityProvider?: IdentityProvider;
	clientTlsIdentity?: string;
	tlsInfo?: {
		certificate: string;
		key: string;
	};
	discovery?: DiscoveryOptions;
	eventHandlerOptions?: DefaultEventHandlerOptions;
	queryHandlerOptions?: DefaultQueryHandlerOptions;
	'connection-options'?: any;
}

export interface ConnectedGatewayOptions extends GatewayOptions {
	discovery: DiscoveryOptions;
	eventHandlerOptions: DefaultEventHandlerOptions;
	queryHandlerOptions: DefaultQueryHandlerOptions;
}

export interface DiscoveryOptions {
	asLocalhost?: boolean;
	enabled?: boolean;
}

export interface DefaultEventHandlerOptions {
	commitTimeout?: number;
	endorseTimeout?: number;
	strategy?: TxEventHandlerFactory | null;
}

export interface DefaultQueryHandlerOptions {
	strategy?: QueryHandlerFactory;
	timeout?: number;
}